pkg/idtools/idtools_unix.go

```diff
@@ -218,61 +88,3 @@ func setPermissions(p string, mode os.FileMode, owner Identity, stat os.FileInfo
 	}
 	return os.Chown(p, owner.UID, owner.GID)
 }
-
-// LoadIdentityMapping takes a requested username and
-// using the data from /etc/sub{uid,gid} ranges, creates the
-// proper uid and gid remapping ranges for that user/group pair
-func LoadIdentityMapping(name string) (IdentityMapping, error) {
```

Comment:
Can you add a `idtools_deprecated_unix.go` file with aliases for the exported functions,  and deprecate those, or won't that work due to a circular import in that case? (we might still be able to after `idtools` is moved to a separate module.


```go
// LoadIdentityMapping takes a requested username and
// using the data from /etc/sub{uid,gid} ranges, creates the
// proper uid and gid remapping ranges for that user/group pair.
//
// Deprecated: this function was only used internally and will be removed in the next release.
func LoadIdentityMapping(name string) (IdentityMapping, error) {
	return usergroup. LoadIdentityMapping(name)
}
```

Reply:
Can add back later if we need. Who is that deprecation message for though? I think we could have documentation about what is breaking with each release but if we require stretching every refactor to a two release cycle then we won't it will make doing all the valuable way more time consuming. IMO we are already doing major version bumps and we should cleanup as much as we can in v28, switch to go.mod, then actually semver the go sdk.

Reply:
> Who is that deprecation message for though?

Always hard to tell.

<img width="675" alt="Screenshot 2024-12-13 at 18 00 04" src="https://github.com/user-attachments/assets/daa4392a-1356-4d07-a3cf-5f28f803d7e2" />


The `Deprecated` annotations are effectively the documentation here, and those _are_ picked up by linters and IDEs, so help users find their replacement. Updating references is not always under control by the user, because go modules may force them to use dependencies that perhaps have not yet switched over. We for sure won't be able to do this for _all_ changes, but if it's only adding a file with aliases, that's not a ton of things to maintain, and can go a long way not having to deal with the aftermath.

Reply:
☝️ OK: identified at least one; BuildKit breaks because  of this;

```
0.137 + xx-go build '-gcflags=' -ldflags '-X github.com/moby/buildkit/version.Version=v0.18.0-55-g80600495c -X github.com/moby/buildkit/version.Revision=80600495cbc8c0c439d71fc77f146cb26fabcfa5 -X github.com/moby/buildkit/version.Package=github.com/moby/buildkit -extldflags '"'"'-static'"'" -tags 'osusergo netgo static_build seccomp ' -o /usr/bin/buildkitd ./cmd/buildkitd
71.51 # github.com/moby/buildkit/cmd/buildkitd
71.51 cmd/buildkitd/util_linux.go:25:27: undefined: idtools.LoadIdentityMapping
```


- see https://github.com/moby/buildkit/pull/5602#discussion_r1895684660